RecoTauTag/RecoTau/src/PositionAtECalEntranceComputer.cc

```diff
@@ -2,37 +2,131 @@
 
 #include "MagneticField/Engine/interface/MagneticField.h"
 #include "MagneticField/Records/interface/IdealMagneticFieldRecord.h"
+#include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+#include "Geometry/Records/interface/CaloGeometryRecord.h"
 #include "CommonTools/BaseParticlePropagator/interface/BaseParticlePropagator.h"
+#include "DataFormats/GeometrySurface/interface/Plane.h"
+#include "TrackPropagation/RungeKutta/interface/defaultRKPropagator.h"
 
 #include <cassert>
 
-PositionAtECalEntranceComputer::PositionAtECalEntranceComputer() : bField_z_(-1.) {}
+//HGCal helper classes
+//MB: looks be copy of HGCal utils: L1Trigger/L1THGCalUtilities/plugins/ntuples/HGCalTriggerNtupleGen.cc
+namespace HGCal_helpers {
+  class SimpleTrackPropagator {
```

Comment:
what's wrong with the usual `PropagatorWithMaterial` (it's using RK as well and has already supported interface and used in other places)?

@bendavid @hatakeyamak 
what is (expected) to be used in the PF linking or 
 @afiqaize @SohamBhattacharya 
which propagation is used to get from a track to the HGCAL in the EGM-specific code?
We should use something consistent.

Reply:
I think that there is nothing particularly wrong, simply this one was used by developers as advised by some phase2 experts. I will be happy to change and test  `PropagatorWithMaterial` if it is preferred - an example of its usage will be helpful.

Reply:
I think use of existing tools is preferred in general.
if I'm not mistaken the interface is there to use the default PF `BaseParticlePropagator` :
call `setPropagationConditions` (I guess radius can be very large or some 400) first and then `propagate`.